WordPress/WordPressTest/MediaEditor/MediaEditorTests.swift

```diff
@@ -6,71 +6,97 @@ import Nimble
 
 class MediaEditorTests: XCTestCase {
 
+    private var cropViewControllerMock: TOCropViewControllerMock!
+    private var mediaEditor: MediaEditor!
+
+    override func setUp() {
+        super.setUp()
+        cropViewControllerMock = TOCropViewControllerMock()
+        mediaEditor = MediaEditor(cropViewControllerFactory: cropViewControllerMockFactory)
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        cropViewControllerMock = nil
+        mediaEditor = nil
+    }
+
     func testGiveTheImageToCropViewController() {
         let image = UIImage()
-        let mediaEditor = MediaEditor()
+        var imageToCrop: UIImage?
+        let mediaEditor = MediaEditor(cropViewControllerFactory: { image in
+            imageToCrop = image
+            return self.cropViewControllerMock
+        })
 
-        mediaEditor.edit(image) { _ in }
+        mediaEditor.edit(image, onFinishEditing: { _ in })
 
-        expect(mediaEditor.cropViewController?.image).to(equal(image))
+        expect(imageToCrop).to(equal(image))
     }
 
     func testReturnsTheCroppedImage() {
         let originalImage = UIImage()
         let croppedImage = UIImage()
         var returnedImage: UIImage?
-        let mediaEditor = MediaEditor.init(cropViewControllerFactory: TOCropViewControllerMock.init)
 
-        mediaEditor.edit(originalImage) { croppedImage in
+        mediaEditor.edit(originalImage, onFinishEditing: { croppedImage in
             returnedImage = croppedImage
-        }
-        (mediaEditor.cropViewController as? TOCropViewControllerMock)?.croppedImage = croppedImage
+        })
+        cropViewControllerMock.croppedImage = croppedImage
```

Comment:
I don't think I understand this test. Questions:
1. Shouldn't this be `cropViewControllerMock.croppedImage = returnedImage` ?
2. Is it right that the mediaEditor.edit method is guaranteed to be on the main thread and that is why you are assigning value to  cropViewControllerMock.croppedImage outside of the onFinishEditing block?
3. Why are we assigning value to cropViewControllerMock.croppedImage if we are not using it in the test or did you mean to assert something else?

Reply:
I guess the syntax used was confusing. I changed it (1209d5755d97445f3e6e54f1a95a4a3fee5321f9), let me know what you think.

What this line is doing is simulating that the user has cropped the image and telling to the mock which image it should return.